app/src/components/EditScreenInfo.tsx

```diff
@@ -1,12 +1,29 @@
-import React from 'react';
-import { StyleSheet } from 'react-native';
+import React from "react";
+import { StyleSheet } from "react-native";
 
-import Colors from '../constants/Colors';
-import { ExternalLink } from './ExternalLink';
-import { MonoText } from './StyledText';
-import { Text, View } from './Themed';
+import Colors from "../constants/Colors";
+import { ExternalLink } from "./ExternalLink";
+import { MonoText } from "./StyledText";
+import { Text, View } from "./Themed";
+import { useAuth } from "@/providers/AuthProvider";
+import { useCreateSolanaWallet } from "@/api/orders";
 
 export default function EditScreenInfo({ path }: { path: string }) {
+  // example of an auth context consumer
+  const { session, loading, profile } = useAuth();
+  const { mutateAsync: createSolana } = useCreateSolanaWallet();
```

Comment:
and then here's an example of consuming the auth context in a component, in your routing component you'd want to redirect if they're not logged in. 

and then this should work for to create a solana wallet but i wasnt able to test it. i think i need some  .env configs to get the app working locally.

Reply:
for the `supabase auth is checked on the webserver side` stuff, i think that should be as straightforward as moving most of these supabase auth methods to the backend

```
 await supabase.auth.signInWithPassword({
      email,
      password,
    });
```

and then making them separate endpoints that the frontend hits and stores the session / data  / result etc

Reply:
that or maybe the backend can check the frontend's auth token with `supabase.auth.getUser`? dunno if that works but would be faster than moving everything to the backend